plugin/src/main/java/com/denizenscript/denizen/events/block/BlockExplodesScriptEvent.java

```
@@ -35,12 +34,37 @@ public class BlockExplodesScriptEvent extends BukkitScriptEvent implements Liste
     //
     // @Determine
     // ListTag(LocationTag) to set a new lists of blocks that are to be affected by the explosion.
-    // ElementTag(Decimal) to change the strength of the explosion.
+    // "STRENGTH:<ElementTag(Decimal)>" to change the strength of the explosion.
     //
     // -->
 
     public BlockExplodesScriptEvent() {
         registerCouldMatcher("<block> explodes");
+        this.<BlockExplodesScriptEvent, ObjectTag>registerOptionalDetermination(null, ObjectTag.class, (evt, context, value) -> {
+            String blocks = value.identify();
+            if (blocks.contains(",") || blocks.startsWith("li@")) { // Raw ListTag check due to there not being a prefix for block strength previously
+                evt.event.blockList().clear();
+                for (LocationTag newBlock : value.asType(ListTag.class, context).filter(LocationTag.class, context)) {
+                    if (newBlock.getWorld() != null) {

```

this check shouldn't be here like this -- either don't check (allow an exception to propagate), or check and handle the error. This code as written just silently ignores bad input